### PR TITLE
Fixing websocket subprotocol header validation

### PR DESCRIPTION
**Pull request by jmr0**
This takes care of https://github.com/servo/servo/issues/9034

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9400)

### CODE REVIEW

**Comment by KiChjang** [comment]
This currently fails to build:
```
/home/travis/build/servo/servo/components/script/dom/websocket.rs:222:52: 222:82 error: no method named `eq_ignore_ascii_case` found for type `&collections::string::String` in the current scope

/home/travis/build/servo/servo/components/script/dom/websocket.rs:222             if protocols[i + 1..].iter().any(|p| p.eq_ignore_ascii_case(protocol)) {

                                                                                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/home/travis/build/servo/servo/components/script/dom/websocket.rs:222:52: 222:82 help: items from traits can only be used if the trait is in scope; the following trait is implemented but not in scope, perhaps add a `use` for it:

/home/travis/build/servo/servo/components/script/dom/websocket.rs:222:52: 222:82 help: candidate #1: use `std::ascii::AsciiExt`

error: aborting due to previous error

Could not compile `script`.
```

**Comment by nox** [comment]
There are no new tests passing, could you write one that fail without this new code?

**Comment by jmr0** [comment]
Added 2 tests to wpt that will fail without these changes. Thanks.

**Comment by KiChjang** [comment]
Did you use the `./mach create-wpt` command to create the tests? Travis CI is complaining.

File: components/util/str.rs [hunk 1]
```diff
@@ -565,3 +565,37 @@ pub fn search_index(index: usize, indices: CharIndices) -> isize {
...
+            34 |
+            47 |
+            91 |
+            93 |
+            63 |
+            61 |
+            123 |
+            125 |
+            32 => false, // separators
+            x if x > 127 => false, // non-CHARs
```
Comment (KiChjang) [inline components/util/str.rs]: This could be `127.. => false` instead.

**Comment by KiChjang** [comment]
Looks good to me after my small nit. @nox do the tests look good to you?

File: components/util/str.rs (hunk 1 above)
Comment (frewsxcv) [inline components/util/str.rs]: Open ranges in matching is not supported yet

https://github.com/rust-lang/rfcs/issues/947

**Comment by jdm** [comment]
Thanks @jmr0!

**Pull request merged**